private/buf/buffetch/ref_parser.go

```diff
@@ -49,18 +49,29 @@ func newRefParser(logger *zap.Logger) *refParser {
 		fetchRefParser: internal.NewRefParser(
 			logger,
 			internal.WithRawRefProcessor(processRawRef),
-			internal.WithSingleFormat(formatBin),
-			internal.WithSingleFormat(formatBinpb),
+			internal.WithSingleFormat(
+				formatBin,
+				internal.WithSingleCustomOptionKey(validateKey),
```

Comment:
Validate shouldn't be on anything except yaml right?

Reply:
You can validate any message, after it has been decoded. `protoyaml` just provide better errors if you give it the validator directly.